### PR TITLE
Stop travis build if the only files changed are doc files

### PR DESCRIPTION
**Pull request by sehqlr**
Brian suggested this change. This is more automated than adding
ci skip to commit messages

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -11,6 +11,7 @@ env:
 install: "pip install -r requirements.txt"
 
 before_script:
+  - if [ "git diff --name-only $TRAVIS_COMMIT_RANGE | grep -v -e 'md$' -e 'rst$'" == "" ]; then return 1; fi
```
Comment (siddharthist) [inline .travis.yml]: Won't this just evaluate that as a string, not as a command to be run?

Comment (sehqlr) [inline .travis.yml]: Damn it, you are right.

File: .travis.yml [hunk 2]
```diff
@@ -11,6 +11,7 @@ env:
 install: "pip install -r requirements.txt"
 
 before_script:
+  - if [ "$(git diff --name-only $TRAVIS_COMMIT_RANGE | grep -v -e 'md$' -e 'rst$')" == "" ]; then return 1; fi
```
Comment (BrianHicks) [inline .travis.yml]: will this work with single `[`? How about `[ -z "$(...)" ]` instead?

Comment (BrianHicks) [inline .travis.yml]: also, doesn't this exit with a non-zero exit code? Wouldn't this fail the build?

**Comment by sehqlr** [comment]
@BrianHicks I read your comment about the build failing, and you're right, that's what it'd do. Where should I put the logic? In the build section?

**Comment by siddharthist** [comment]
Closing this in favor of #1249

**Pull request closed without merging**